yt/visualization/plot_modifications.py

```diff
@@ -2492,15 +2490,14 @@ def __call__(self, plot):
                         "'time_offset' must be a float, tuple, or" "YTQuantity!"
                     )
                 t -= toffset.in_units(self.time_unit)
-            if isinstance(self.time_unit, Unit):
+            try:
                 # here the time unit will be in brackets on the annotation.
-                # This will most likely be in "code_time".
                 un = self.time_unit.latex_representation()
                 time_unit = r"$\ \ (" + un + r")$"
-            else:
-                # the 'smallest_appropriate_unit' function will return a
-                # string, so we shouldn't need to format it further.
-                time_unit = self.time_unit
+            except AttributeError:
+                # this should only happen if self.time_unit is "code_time"
+                # in which case we drop the "_" in the time stamp
+                time_unit = str(self.time_unit).replace("_", " ")
```

Comment:
Just to make sure we're not accidentally hitting this in any unforeseen way.
```suggestion
                # in which case we drop the "_" in the time stamp
                assert plot.ds.unit_system._code_flag == "code"
                time_unit = str(self.time_unit).replace("_", " ")
```

Reply:
This is still relevant but I know have a better idea